feature_tests/kotlin/somelib/src/main/kotlin/dev/diplomattest/somelib/TraitWrapper.kt

```diff
@@ -0,0 +1,46 @@
+package dev.diplomattest.somelib
+
+import com.sun.jna.Callback
+import com.sun.jna.Library
+import com.sun.jna.Native
+import com.sun.jna.Pointer
+import com.sun.jna.Structure
+
+internal interface TraitWrapperLib: Library {
+    fun TraitWrapper_test_with_trait(t: DiplomatTrait_TesterTrait_Wrapper_Native, x: Int): Int
+    fun TraitWrapper_test_trait_with_struct(t: DiplomatTrait_TesterTrait_Wrapper_Native): Int
+}
+
+internal class TraitWrapperNative: Structure(), Structure.ByValue {
+    @JvmField
+    internal var cantBeEmpty: Byte = 0;
+  
+    // Define the fields of the struct
+    override fun getFieldOrder(): List<String> {
+        return listOf("cantBeEmpty")
+    }
+}
+
+class TraitWrapper internal constructor (
+    internal val nativeStruct: TraitWrapperNative) {
+    val cantBeEmpty: Boolean = nativeStruct.cantBeEmpty > 0
+
+    companion object {
+        internal val libClass: Class<TraitWrapperLib> = TraitWrapperLib::class.java
+        internal val lib: TraitWrapperLib = Native.load("somelib", libClass)
+        val NATIVESIZE: Long = Native.getNativeSize(TraitWrapperNative::class.java).toLong()
+        
+        fun testWithTrait(t: DiplomatTrait_TesterTrait_Wrapper, x: Int): Int {
+            
+            val returnVal = lib.TraitWrapper_test_with_trait(t.nativeStruct, x);
+            return returnVal
+        }
+        
+        fun testTraitWithStruct(t: DiplomatTrait_TesterTrait_Wrapper): Int {
+            
+            val returnVal = lib.TraitWrapper_test_trait_with_struct(t.nativeStruct);
+            return returnVal
+        }
```

Comment:
I think this would be much more ergonomic as a library consumer, and if that's the case you could ignore the comment about the public interface for `DiplomatTrait_TesterTrait_Wrapper.fromTrait`
```suggestion
        fun testTraitWithStruct(t: TesterTrait): Int {
            val tWrapper = DiplomatTrait_TesterTrait_Wrapper.fromTraitObj(t)
            val returnVal = lib.TraitWrapper_test_trait_with_struct(tWrapper.nativeStruct);
            return returnVal
        }
```
        }
```

Reply:
good idea -- I've done this instead, and left the internal wrapper names (for the different trait construction bits) with `_` in the names.